auth-service/src/main/kotlin/ru/zv/authservice/util/PhoneNumberParser.kt

```diff
@@ -0,0 +1,17 @@
+package ru.zv.authservice.util
+
+import com.google.i18n.phonenumbers.PhoneNumberUtil
+import io.grpc.Status
+import ru.zv.authservice.exceptions.AuthException
+import ru.zv.authservice.service.dto.PhoneNumber
+
+object PhoneNumberParser {
+
+    private val phoneNumberUtils = PhoneNumberUtil.getInstance()
+    fun stringToServicePhone(phone: String): PhoneNumber {
+        phone.takeIf { phoneNumberUtils.isPossibleNumber(phone, "RU") }?.let {
+            val parsedPhone = phoneNumberUtils.parse(phone, "RU")
+            return PhoneNumber(parsedPhone.countryCode, parsedPhone.nationalNumber)
+        } ?: throw AuthException("Failed to parse phone number", Status.Code.INVALID_ARGUMENT)
```

Comment:
```suggestion
        phone.takeIf { phoneNumberUtils.isPossibleNumber(phone, "RU") }
            ?.let {
                val parsedPhone = phoneNumberUtils.parse(phone, "RU")
                return PhoneNumber(parsedPhone.countryCode, parsedPhone.nationalNumber)
            }
            ?: throw AuthException("Failed to parse phone number", Status.Code.INVALID_ARGUMENT)
```

возьми в привычку все новые методы писать с новой строки -- это опять же, тупо легче читается, чем все в одну строку